### ISSUE TITLE
Axe accessibility issue with Button inside Antd 4.x Collapse

### ISSUE THREAD

**Issue by abhishekahuja87** [ant-design/ant-design#31122]
I see accessibility issue when using [Axe](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd) accessibility chrome plugin on my App which has [Collapse](https://ant.design/components/collapse/) from Antd 3.x

[codesandbox link](https://codesandbox.io/embed/elastic-colden-odgbw?fontsize=14&hidenavigation=1&theme=dark)

Issue: Nested interactive controls are not announced by screen readers

Reason: Since there is a button in the accordion

If I remove the button, the above accessibility is solved. But I would like to have the button

I also tried to replace button by span like 

    <span tabIndex="0"><Icon/></span> 

but even this way the accessibility issue persists. But if I remove the 'tabIndex="0"', it works fine.

What is the way to put items in the Accordion which are also accessible via keyboard(hence tabIndex=0).

TIA
